### ISSUE TITLE
Ghost n Goblins not loading

### ISSUE THREAD

**Issue by bnice** [dirkwhoffmann/vAmiga#422]
Hi,

Good work on fixing Ruff N Tumble. I am slowly working my through my collection to help test vAmiga :-). Next game I have found a problem with is Ghost n Goblins make sure to load with 1/2 mb chip and 1/2 mb slow. On vAmiga it freezes on track 5. It works fine on FS-UAE 3.0.5. Image can be found https://www.planetemu.net/index.php?section=roms&action=showrom&datSlug=commodore-amiga-games-adf&romSlug=ghosts-n-goblins-1988-elite-10& 

Regards,

Ben.

**Comment by dirkwhoffmann** [comment]
Can you retry with "Asynchronous FIFO" enabled (which is the setting that is supposed to be most accurate)?

<img width="645" alt="Bildschirmfoto 2020-10-13 um 18 53 25" src="https://user-images.githubusercontent.com/12561945/95891565-8a7a1480-0d85-11eb-85bb-b7fbe639d5d4.png">

Interestingly, it also boots fine with a Turbo Drive. This means I better check why it fails to load with the simplified FIFO implementation...

**Comment by bnice** [comment]
Thanks, boots fine with async FIFO. Will test with this setting, if I get a failure in  future.

**Comment by dirkwhoffmann** [comment]
> Interestingly, it also boots fine with a Turbo Drive

Here, I fooled myself. When I did the test with a Turbo Drive, I was using a different version of Ghost n Goblins 🙄. Your version freezes with a Turbo drive, too.

I also found the reason. Here is the debug log with "Async FIFO" disabled.

```
...
[1276] (179,  9) FC0F94  0 BCBSD- 606E 1000 DiskController: SYNC IRQ (dsklen = 34143) 4489
[1277] (164,  7) FC0F94  0 BCBSD- 606E 1000 DiskController: SYNC IRQ (dsklen = 33250) 4489
[1277] (164,  9) FC0F94  0 BCBSD- 606E 1000 DiskController: SYNC IRQ (dsklen = 33249) 4489
[1278] ( 11,  9) FC0E9C  0 BCBSD- 606E 1002 DiskController: DRIVE_DMA_READ -> DRIVE_DMA_OFF
```

Now, when I enable "Async FIFO", the game continues to load: 

```
[18840] (224,180) 040226  0 Bcbsd- 206C 0020 DiskController: OPT_ASYNC_FIFO 1
[18840] (257,112) 04022C  0 Bcbsd- 206C 0020 DiskController: SYNC IRQ (dsklen = 0) 4489
[18840] (257,224) 040212  0 Bcbsd- 206C 1020 DiskController: SYNC IRQ (dsklen = 0) 4489
...
```

The game loader requires DSKSYNC interrupts to happed even when DMA is off. If "Async FIFO" is disabled, these interrupts may only occur when something is read from the disk, because this is the place where disk rotation is emulated then. If "Async FIFO" is enabled, disk rotation is emulated asynchronously and the occurrence of DSKSYN interrupts is independent from DMA status. 

Bottom line: It's not a bug, it's just a limitation of synchronous FIFO mode.
